fedimint-core/src/api.rs

```diff
@@ -905,6 +905,32 @@ pub struct ConsensusContribution {
     pub time: SystemTime,
 }
 
+/// The state of the server returned via APIs
+#[derive(Debug, Clone, Default, Serialize, Deserialize, Eq, PartialEq)]
+pub enum ServerStatus {
+    /// Server needs a password to read configs
+    #[default]
+    AwaitingPassword,
+    /// Waiting for peers to share the config gen params
+    SharingConfigGenParams,
+    /// Ready to run config gen once all peers are ready
+    ReadyForConfigGen,
+    /// We failed running config gen
+    ConfigGenFailed,
+    /// Config is generated, peers should verify the config
+    VerifyingConfigs,
+    /// Restarted from a planned upgrade (requires action to start)
+    Upgrading,
+    /// Consensus is running
+    ConsensusRunning,
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq)]
+pub struct StatusResponse {
+    pub server: ServerStatus,
+    pub consensus: Option<ConsensusStatus>,
+}
```

Comment:
Nit: Would it make sense to eventually merge these two structs? E.g. `ConsensusStatus` could be a member of `ServerStatus::ConsensusRunning`. Other states could also carry additional info (e.g. error messages for config gen failed).

Reply:
This was tried and the resulting serialized json is very hard to parse. I works if you're only using rust, but if you are interoperating with other system you need to simplify the result.